File: src/main/java/uk/co/ben_gibson/git/link/Git/Commit.java
package uk.co.ben_gibson.git.link.Git;

import org.jetbrains.annotations.NotNull;

public class Commit
{
    private final String hash;

    public Commit(@NotNull final String hash) {
        this.hash = hash;
    }

    @NotNull
    public String shortHash()
    {
        return hash.substring(0, 6);
    }

    @NotNull
    public String hash()
    {
        return hash;
    }
}

File: src/main/kotlin/uk/co/ben_gibson/git/link/git/Commit.kt
package uk.co.ben_gibson.git.link.git

data class Commit(private val hash: String) {
    init {
        require(hash.isNotBlank()) { "Commit hash cannot be blank" }
        require(hash.length >= 6) { "Commit hash must be at least 6 characters" }
    }

    val shortHash: String
        get() = hash.substring(0, minOf(6, hash.length))

    override fun toString(): String = hash
}